File: content/problems/0082-remove-duplicates-from-sorted-list-ii/solutions/dummy-and-groups/solution.cpp
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode dummy(0, head); ListNode* previous = &dummy; ListNode* current = head;
        while (current) { if (current->next && current->val == current->next->val) { int value = current->val; while (current && current->val == value) current = current->next; previous->next = current; } else { previous = current; current = current->next; } } return dummy.next;
    }
};

File: content/problems/0082-remove-duplicates-from-sorted-list-ii/solutions/dummy-and-groups/Solution.kt
class Solution {
    fun deleteDuplicates(head: ListNode?): ListNode? {
        val dummy = ListNode(0); dummy.next = head; var previous: ListNode = dummy; var current = head
        while (current != null) { if (current.next != null && current.`val` == current.next!!.`val`) { val value = current.`val`; while (current != null && current.`val` == value) current = current.next; previous.next = current } else { previous = current; current = current.next } }
        return dummy.next
    }
}
